app/controllers/api/v1/photos_controller.rb

```
@@ -20,9 +20,16 @@ def thumbnail
   private
 
   def fetch_cached_thumbnail(source)
-    Rails.cache.fetch("photo_thumbnail_#{params[:id]}", expires_in: 1.day) do
-      Photos::Thumbnail.new(current_api_user, source, params[:id]).call
+    cached = Rails.cache.read("photo_thumbnail_#{params[:id]}")
+    return cached if cached
+
+    response = Photos::Thumbnail.new(current_api_user, source, params[:id]).call
+
+    if response.success?
```

RuboCop (1.82.1)

[convention] Favor modifier `if` usage when having a single-line body. Another good alternative is the usage of control flow `&&`/`||`.

(Style/IfUnlessModifier)